src/User.php

```diff
@@ -7013,4 +7013,31 @@ private function cleanInput(array $input): array
 
         return $input;
     }
+
+    /**
+     * User has right for given module and right.
+     *
+     * @param string  $module Module to check
+     * @param integer $right  Right to check
+     *
+     * @return boolean|int
+     **/
+    public function hasRight($module, $right)
+    {
+        if ($this->isNewItem()) {
+            throw new \LogicException('Cannot check rights for a user not yet saved');
+        } else {
+            $user_id = $this->getID();
+            $profiles = Profile_User::getUserProfiles($user_id);
+            foreach ($profiles as $profile_id) {
+                $profile = new Profile();
+                $profile->getFromDB($profile_id);
+                $profile->cleanProfile();
+                if ($profile->haveUserRight($user_id, $module, $right, 0)) {
```

Comment:
Yous should check the user rights on the entity of the ITIL object. Indeed, the user may not have the right to see private items on the root entity, but have rights to view them in a sub-entity.